### PR TITLE
Openid4vp i18n

### PR DESCRIPTION
**Pull request by beardedfoo**
This adds a localize() function which can be used to localize the template path used in response rendering

### CODE REVIEW

File: auth/api/iam/assets/authz_wallet_nl.html [hunk 1]
```diff
@@ -0,0 +1,60 @@
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Credential vereist</title>
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.3/dist/css/bootstrap.min.css" integrity="sha384-rbsA2VBKQhggwzxH7pPCaAqO46MgnOM80zW1RWuH61DGLwZJEdK2Kadq2F9CUG65" crossorigin="anonymous">
+</head>
+<body>
+    <h1>Credential vereist (Wallet)</h1>
+    <p>{{ .VerifierName }} vraagt om identificatie met behulp van een of meer credentials in uw wallet:</p>
```
Comment (reinkrul) [inline auth/api/iam/assets/authz_wallet_nl.html:9]:
```suggestion
    <p>{{ .VerifierName }} vraagt om identificatie met één of meer credentials in uw wallet:</p>
```

File: auth/api/iam/localize.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+// localize selects from a range of localized values based on the accept-language header, returning
+// a default value if no localized value matches the languages specified in the accept-language header.
+func localize(defaultValue string, localizedValues map[string]string, acceptLanguageHeader string) string {
+	// Iterate over the accepted languages specified by the user-agent
+	for _, acceptedLanguage := range strings.Split(acceptLanguageHeader, ",") {
+		// Trim any whitespace from the language specification
+		acceptedLanguage = strings.TrimSpace(acceptedLanguage)
+
+		// Split the language and any associated q-factor value (e.g. "en;q=0.5") where the q-factor specifies
```
Comment (reinkrul) [inline auth/api/iam/localize.go:15]: does anyone actually implement that properly... talking about elaborate specs

File: auth/api/iam/authorized_code.go [hunk 3]
```diff
@@ -38,7 +39,7 @@ func (a authorizedCodeFlow) handleAuthzRequest(params map[string]string, session
 
 	// Render HTML
 	buf := new(bytes.Buffer)
-	// TODO: Support multiple languages
+	// TODO: Support multiple languages -- but how with the template path hardcoded prior to the request?
```
Comment (reinkrul) [inline auth/api/iam/authorized_code.go:42]: this authorizedCodeFlow type will probablyt go away, it's a reminiscant of the previous, complicated, OAuth2 design I made earlier

File: auth/api/iam/localize.go [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package iam
+
+import (
+	"strings"
+)
+
+// localize selects from a range of localized values based on the accept-language header, returning
+// a default value if no localized value matches the languages specified in the accept-language header.
+func localize(defaultValue string, localizedValues map[string]string, acceptLanguageHeader string) string {
```
Comment (reinkrul) [inline auth/api/iam/localize.go:9]: I guess browsers always send their preferred language?

**Review by reinkrul — CHANGES_REQUESTED** [review]

File: auth/api/iam/localize.go (hunk 2 above)
Comment (beardedfoo) [reply auth/api/iam/localize.go:15]: Yea this is asking a lot of implementers. It wouldn't be *too* insane to create a map of decimal values and then sort the keys based on them but what would you put as a default value when the q-factor is missing?

File: auth/api/iam/localize.go (hunk 4 above)
Comment (beardedfoo) [reply auth/api/iam/localize.go:9]: Never mix the words "browser" and "always" in a sentence. We will just fall back to default if it is blank, which I should create a unit test for.

**Review by reinkrul — APPROVED** [review]

**Pull request merged**